src/Evolution/DiscontinuousGalerkin/Actions/ApplyBoundaryCorrections.hpp

```diff
@@ -472,13 +522,19 @@ bool receive_boundary_data_local_time_stepping(
                 mortar_id, received_mortar_data->second.volume_mesh);
             neighbor_mortar_data.face_mesh =
                 received_mortar_data->second.interface_mesh;
-            neighbor_mortar_data.mortar_mesh = mortar_meshes.at(mortar_id);
+            neighbor_mortar_data.mortar_mesh =
+                received_mortar_data->second.boundary_correction_mesh.value();
             neighbor_mortar_data.mortar_data = std::move(
                 received_mortar_data->second.boundary_correction_data.value());
             boundary_data_history->at(mortar_id).remote().insert(
                 mortar_next_time_step_id,
                 received_mortar_data->second.integration_order,
                 std::move(neighbor_mortar_data));
+            if (mortar_mesh !=
+                received_mortar_data->second.boundary_correction_mesh) {
+              boundary_data_history->at(mortar_id).remote().for_each(
+                  project_boundary_mortar_data);
```

Comment:
Again, clear the cache, unless this is only modifying the entry that was just inserted, in which case add a comment that you don't have to clear the cache.